src/app/auth/resetPassword/index.tsx

```diff
@@ -0,0 +1,250 @@
+import { ScrollView } from 'react-native-gesture-handler';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import validator from 'validator';
+
+import styles from './styles';
+import Icon from '../../../../assets/icons';
+import StyledButton from '../../../components/StyledButton/StyledButton';
+import UserStringInput from '../../../components/UserStringInput/UserStringInput';
+import colors from '../../../styles/colors';
+import globalStyles from '../../../styles/globalStyles';
+import { useSession } from '../../../utils/AuthContext';
+import supabase from '../../../utils/supabase';
+
+function ResetPasswordScreen() {
+  const { updateUser, signOut } = useSession();
+  const [password, setPassword] = useState('');
+  const [passwordTextHidden, setPasswordTextHidden] = useState(true);
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordTextHidden, setConfirmPasswordTextHidden] =
+    useState(true);
+  const [loading, setLoading] = useState(false);
+
+  const [hasUppercase, setHasUppercase] = useState(false);
+  const [hasLowercase, setHasLowercase] = useState(false);
+  const [hasNumber, setHasNumber] = useState(false);
+  const [hasLength, setHasLength] = useState(false);
+  const [isDifferent, setIsDifferent] = useState(false);
+  const [isMatching, setIsMatching] = useState(false);
+
+  const checkPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      if (text !== text.toLowerCase()) {
+        setHasUppercase(true);
+      } else {
+        setHasUppercase(false);
+      }
+      if (text !== text.toUpperCase()) {
+        setHasLowercase(true);
+      } else {
+        setHasLowercase(false);
+      }
+      if (/[0-9]/.test(text)) {
+        setHasNumber(true);
+      } else {
+        setHasNumber(false);
+      }
+      if (text.length >= 8) {
+        setHasLength(true);
+      } else {
+        setHasLength(false);
+      }
+      //need to check that it is different from old password
+    }
+  };
+
+  const checkConfirmPassword = (text: string) => {
+    setConfirmPassword(text);
+    if (text !== '') {
+      if (text == password) {
+        setIsMatching(true);
+      } else {
+        setIsMatching(false);
+      }
+    }
+  };
+
+  const changePassword = async () => {
+    setLoading(true);
+    const { error } = await updateUser({ password });
+
+    if (error) {
+      Alert.alert('Updating password failed');
+    } else {
+      await signOut();
+      router.replace('/auth/login');
+    }
+
+    setLoading(false);
+  };
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <Text style={[globalStyles.h1]}>Create a New Password</Text>
+        <View style={styles.newPassword}>
+          <UserStringInput
+            placeholder="Password"
+            label="New password"
+            placeholderTextColor={colors.darkGrey}
+            onChange={text => {
+              setPassword(text);
+              checkPassword(text);
```

Comment:
You should also run `checkConfirmPassword` here because if you change this text box when the inputs were previously equal, the UI will show that the inputs still match because `checkConfirmPassword` is not being run.